### PR TITLE
Quantity unit using dimensions

### PR DESCRIPTION
**Pull request by hpohekar**
Now we can construct quantity using following ways - 

1.  `velocity = q.Quantity(16.6, unit="m s^-1")`

2. `velocity = q.Quantity(16.6, quantity_map={"Velocity": 1.0})`

3. `velocity = q.Quantity(16.6, dimensions=[0.0, 1.0, -1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])`

4. `velocity = q.Quantity(16.6, dimensions=[0, 1, -1])`

5.  `velocity = q.Quantity(16.6, dimensions=[0.0, 1.0, -1.0])`

### CODE REVIEW

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/quantity.py [hunk 1]
```diff
@@ -868,12 +862,19 @@ class Quantity(float):
...
+            raise ValueError("unit or quantity_map or dimensions is allowed.")
         if quantity_map:
             unit_str = get_unit_from_map(quantity_map)
+        if dimensions:
+            if len(dimensions) > 9:
+                raise ValueError("Number of dimensions should be <= 9")
+            dims_to_float = [float(dim) for dim in dimensions]
+            while len(dims_to_float) < 9:
+                dims_to_float.append(0.0)
+            unit_str = get_si_unit_from_dim(dim_list=dims_to_float)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity.py:877]:
```suggestion
            unit_str = get_si_unit_from_dim(dim_list=dims_to_float + ((9-len(dims_to_float))*[0]))
```

File: src/ansys/fluent/core/quantity.py [hunk 2]
```diff
@@ -882,11 +883,18 @@ def __init__(self, real_value, unit_str=None, quantity_map=None):
...
             raise ValueError("Either unit or quantity_map is allowed.")
         if quantity_map:
             unit_str = get_unit_from_map(quantity_map)
+        if dimensions:
+            if len(dimensions) > 9:
+                raise ValueError("Number of dimensions should be <= 9")
+            dims_to_float = [float(dim) for dim in dimensions]
+            while len(dims_to_float) < 9:
+                dims_to_float.append(0.0)
+            unit_str = get_si_unit_from_dim(dim_list=dims_to_float)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity.py:897]:
```suggestion
            unit_str = get_si_unit_from_dim(dim_list=dims_to_float + ((9-len(dims_to_float))*[0]))
```
Alternatively you could do all the padding inside `get_si_unit_from_dim`

File: src/ansys/fluent/core/quantity.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py]: Done. Thank you.

File: src/ansys/fluent/core/quantity.py (hunk 2 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py]: Done. Thank you.

**Review by mkundu1 — APPROVED** [review]

**Pull request merged**